src/wanderer/guzzlr.ts

```diff
@@ -0,0 +1,101 @@
+import { buy, craftType, Location, print, retrieveItem } from "kolmafia";
+import { $item, get, Guzzlr, have } from "libram";
+import { freeCrafts, globalOptions } from "../lib";
+import { garboValue } from "../session";
+import {
+  canAdventureOrUnlock,
+  DraggableFight,
+  WandererTarget,
+  wandererTurnsAvailableToday,
+} from "./lib";
+
+function considerAbandon(locationSkiplist: Location[]) {
+  const location = Guzzlr.getLocation();
+  const remaningTurns = Math.ceil(
+    (100 - get("guzzlrDeliveryProgress")) / (10 - get("_guzzlrDeliveries"))
+  );
+
+  print(
+    `Got guzzlr quest ${Guzzlr.getTier()} at ${Guzzlr.getLocation()} with remaining turns ${remaningTurns}`
+  );
+
+  if (
+    // consider abandoning
+    !location || // if mafia faled to track the location correctly
+    locationSkiplist.includes(location) ||
+    !canAdventureOrUnlock(location) || // or the zone is marked as "generally cannot adv"
+    (globalOptions.ascending && wandererTurnsAvailableToday(location) < remaningTurns) // or ascending and not enough turns to finish
+  ) {
+    print("Abandoning...");
+    Guzzlr.abandon();
+  }
+}
+
+function acceptGuzzlrQuest(locationSkiplist: Location[]) {
+  if (Guzzlr.isQuestActive()) considerAbandon(locationSkiplist);
+  while (!Guzzlr.isQuestActive()) {
```

Comment:
This `while` will always terminate because you can always accept a bronze quest, right?